internal/setup/warning.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/krew/internal/environment"
+	"sigs.k8s.io/krew/internal/installation"
+)
+
+const (
+	instructionWindows = `To be able to run kubectl plugins, you need to add the
+"%USERPROFILE%\.krew\bin" directory to your PATH environment variable
+and restart your shell.`
+	instructionUnixTemplate = `To be able to run kubectl plugins, you need to add
+the following to your %s
+
+and restart your shell.`
+	instructionZsh = `~/.zshrc:
+
+    export PATH="${PATH}:${HOME}/.krew/bin"`
+	instructionBash = `~/.bash_profile or ~/.bashrc:
+
+    export PATH="${PATH}:${HOME}/.krew/bin"`
+	instructionFish = `config.fish:
+
+    set -gx PATH $PATH $HOME/.krew/bin`
+	instructionGeneric = `~/.bash_profile, ~/.bashrc, or ~/.zshrc:
+
+    export PATH="${PATH}:${HOME}/.krew/bin"`
+)
+
+func WarnIfBinDirIsNotInPATH(w io.Writer, paths environment.Paths) {
+	if isBinDirInPATH(paths) {
+		klog.V(4).Info("PATH contains krew's bindir or this is the first install")
+		return
+	}
+
+	boldRed := color.New(color.FgRed, color.Bold).SprintfFunc()
+	fmt.Fprintf(w, "%s: %s\n\n", boldRed("WARNING"), msgForShell())
+}
+
+func msgForShell() string {
+	if installation.IsWindows() {
+		return instructionWindows
+	}
+
+	var instruction string
+	switch shell := os.Getenv("SHELL"); {
+	case strings.HasSuffix(shell, "/zsh"):
+		instruction = instructionZsh
+	case strings.HasSuffix(shell, "/bash"):
+		instruction = instructionBash
+	case strings.HasSuffix(shell, "/fish"):
+		instruction = instructionFish
+	default:
+		instruction = instructionGeneric
+	}
+	return fmt.Sprintf(instructionUnixTemplate, instruction)
+}
+
+func isBinDirInPATH(paths environment.Paths) bool {
+	// For the first run we don't want to show a warning.
+	_, err := os.Stat(paths.BasePath())
+	if err != nil {
+		klog.V(4).Info("Assuming this is the first run")
+		return os.IsNotExist(err)
+	}
+
+	krewPluginPath := paths.BinPath()
+	return strings.Contains(os.Getenv("PATH"), krewPluginPath)
```

Comment:
Let's use if there's an utility in kubectl plugin PATH scanning logic.

If not, please separate by `os.PathListSeparator`, and do a `==` check; not `Contains`.

Reply:
Doesn't really look like they have a dedicated util for that: https://github.com/kubernetes/kubernetes/blob/5cc945ad0d2fad2bdde7591490fcf0401fa335aa/staging/src/k8s.io/kubectl/pkg/cmd/plugin/plugin.go#L105

But checking for equality sounds right.